Run `npm install` as part of `make build`

I recently rebased against main, and now running `make build` fails with a TypeScript error due to an invalid `--target` value passed to the Rollup TypeScript plugin:

---

```sh
make build                                                                                                                                                   ok 
npm run build -w translate

> build
> rollup -c


src/index.tsx → dist/translate.js...
(!) Plugin typescript: @rollup/plugin-typescript TS6046: Argument for '--target' option must be: 'es3', 'es5', 'es6', 'es2015', 'es2016', 'es2017', 'es2018', 'es2019', 'es2020', 'es2021', 'es2022', 'esnext'.
[!] (plugin typescript) RollupError: @rollup/plugin-typescript: Couldn't process compiler options
    at error (/Users/jahu/Documents/Mozilla/pontoon/node_modules/rollup/dist/shared/rollup.js:353:30)
    at Object.error (/Users/jahu/Documents/Mozilla/pontoon/node_modules/rollup/dist/shared/rollup.js:1721:20)
    at emitParsedOptionsErrors (file:///Users/jahu/Documents/Mozilla/pontoon/node_modules/@rollup/plugin-typescript/dist/es/index.js:348:17)
    at Object.buildStart (file:///Users/jahu/Documents/Mozilla/pontoon/node_modules/@rollup/plugin-typescript/dist/es/index.js:746:13)
    at /Users/jahu/Documents/Mozilla/pontoon/node_modules/rollup/dist/shared/rollup.js:1914:40
    at async Promise.all (index 0)
    at PluginDriver.hookParallel (/Users/jahu/Documents/Mozilla/pontoon/node_modules/rollup/dist/shared/rollup.js:1842:9)
    at /Users/jahu/Documents/Mozilla/pontoon/node_modules/rollup/dist/shared/rollup.js:27151:13
    at catchUnfinishedHookActions (/Users/jahu/Documents/Mozilla/pontoon/node_modules/rollup/dist/shared/rollup.js:26401:16)
    at rollupInternal (/Users/jahu/Documents/Mozilla/pontoon/node_modules/rollup/dist/shared/rollup.js:27148:5)


npm error Lifecycle script `build` failed with error:
npm error code 1
npm error path /Users/jahu/Documents/Mozilla/pontoon/translate
npm error workspace translate
npm error location /Users/jahu/Documents/Mozilla/pontoon/translate
npm error command failed
npm error command sh -c rollup -c
make: *** [build-translate] Error 1
```

---

--- a/Makefile
+++ b/Makefile
@@ -45,12 +45,12 @@ translate/dist:
 	make build-translate
 .server-build:
 	make build-server
-node_modules:
+node_modules/.package-lock.json: package-lock.json
 	npm install
 
 build: build-translate build-server
 
-build-translate: node_modules
+build-translate: node_modules/.package-lock.json
 	npm run build -w translate
 
 build-server: server-env translate/dist

